_posts/2020-06-01-rails-fejlesztes-dockerben.md

```diff
@@ -0,0 +1,233 @@
+
+A debuggolás használatához még szükségünk lesz 2 gem-re. Ezek felvételéhez elösször írjuk be az alábbi sorokata a Gemfile-ba.
+
+```Gemfile
+#Gemfile
+
+gem 'debase'
+gem 'ruby-debug-ide'
+```  
+
+Az IDE alt+enter lenyomása után felkínálja, hogy újra build-eli az imaget. Használjuk ezt a lehetősége, hogy a gem-ek belekerüljenek az image-be. Esetleg további funkció eléréséhez is szükség lehet további gem-ek telepítésére, ezt a fentihez hasonló módon tehetjük meg.
+
+## Rubymine használat
+
+A Rails keretrendszer lehetőséget biztosít generátorok használatára, amik segítségével egy CRUD-ot megvalósító, tesztekkel ellátott weboldalt hozhatunk létre. Ennek használatával fogjuk kipróbálni a fejlesztőkörnyezet funkcióit. Ahhoz, hogy parancsokat tudjunk futtatni a konténerben, elöször csatlakoznunk kell hozzá. Ezt az alábbi parancsal tehetjük meg.
+
+```sh
+docker-compose exec web bash
+```
+
+Alapvetően a RubyMine terminálja nem kapcsolódik a konténerekhez. Több mód is van, hogy a fejlesztőkörnyezet intergrált terminálját automatikusan hozzákapcsoljuk az általunk választott konténerhez. Például a Services oldalon az attach menüpont használatával vagy a Shell Path átállításával. Az egyszerűség kedvért most manuálisan, a fenti paranccsal csatlakozunk a konténerhez, mivel nem lesz erre gyakran szükség. 
+
+Ezután a konténerben navigáljunk el a projectünk mappájába (/test_app) és már, ki is adhatjunk a parancsot, amivel hozzáadunk alapvető funkcionalitásokat az alkalmazásunkhoz.
+
+```sh
+rails g scaffold ruby_mine name:string points:integer
+```
+
+Ha mindent jól csináltunk, akkor az alábbi sorok jelennek meg a konzolon:
+
+![Imgur](https://i.imgur.com/wDxKCEh.png)
+
+Láthajuk, hogy létrejöttek oldalak, tesztek, valamint egy kontroller is. Főként ezeket fogjuk használni az IDE tesztelés során. A **zöld háromszög** gomb lenyomásával elindítható a webszerver, ami a [localhost:3000](http://localhost:3000/)-es címen elérhető.
+
+![Imgur](https://i.imgur.com/IJgL6cZ.png)
+
+Ezután próbáljuk ki a többi fontosabb eszköz működését az IDE-ben. A debug, miután beállitottuk, hogy docker exec-el legyen használva, egyből használható.
+
+![Imgur](https://i.imgur.com/Dgln72k.png)
+
+Teszteket is könnyedén futhatunk az IDE GUI-ján keresztül. Ehhez navigáljunk egy testeket tartalmazó fájl-hoz, mondjuk **test/controllers/ruby_mines_controller_test.rb**, itt a **jobb egérgomb** lenyomásával megjelennek a kontextusfüggő lehetőségek. Ezekközül válasszuk a **Run Minitest:**-lehetőséget. A tesztek futásának eredményét is megjeleníti a fejesztőkörnyezet. Emellet van lehetőség a tesztek közül csak egyet futttani, valmint debug módban is elindíthatjuk a teszteket.
```

Comment:
Teszteket is könnyedén futtathatunk az IDE GUI-ján keresztül. Ehhez navigáljunk egy teszteket tartalmazó fájlhoz, mondjuk **test/controllers/ruby_mines_controller_test.rb**-hez, itt a **jobb egérgomb** lenyomásával megjelennek a kontextusfüggő lehetőségek. Ezek közül válasszuk a **Run Minitest:**-lehetőséget. A tesztek futásának eredményét is megjeleníti a fejlesztőkörnyezet. Emellett lehetőség van a tesztek közül csak egyet futtatni, valamint debug módban is elindíthatjuk a teszteket.